### ISSUE TITLE
False Colon Violation inside guard and ternary operator

### ISSUE THREAD

**Issue by Lukasz2891** [realm/SwiftLint#2806]
In 0.33.x

```        
guard let requestData = try? JSONSerialization.data(withJSONObject: dictionary, options: .prettyPrinted),
            let storeURL = sandbox ? Self.sandboxStoreURL : Self.productionStoreURL else {
                return ...
        }
```
produces false Colon Violation (```colon```) inside the ternary ```?:``` operator.

**Comment by a-25** [comment]
Hello, @Lukasz2891. Could you please post the whole method and all variables included? I could not understand what kind of variables sandboxStoreURL and productionStoreURL are.
I tried to check like below but it does not compile (Xcode 10.2.1).
```
class TestClass {
    public static var sandboxStoreURL: String? = "sandboxStoreURL"
    public static var productionStoreURL: String? = "productionStoreURL"
    public static var sandbox = false
}

class TestSubclass: TestClass {
    func test() -> String {
        let dict = ["dictKey":"dictValue"]
        guard let requestData = try? JSONSerialization.data(withJSONObject: dict, options: .prettyPrinted),
            let storeURL = sandbox ? Self.sandboxStoreURL : Self.productionStoreURL else {
                return "empty"
        }
        return "\(requestData)\(storeURL)"
    }
}
```

The essence of this bug it `Self` keyword. I tried 0.34 version, replaced `Self` with `self` and bug has gone.

**Comment by Lukasz2891** [comment]
It’s a default extension to a protocol so big „Self” is needed as a not specified type.

See: https://stackoverflow.com/questions/42037852/why-do-self-and-self-sometimes-refer-to-different-types-in-static-functions

**Comment by a-25** [comment]
Hello, @Lukasz2891. 
Made a PR, did it fix your case?

**Comment by Lukasz2891** [comment]
All works ok in 0.34.0, thanks.

**Comment by Lukasz2891** [comment]
```swift
protocol Test {
    static var aaa: Any? { get }
    static var bbb: Any? { get }
}

extension Test {
    func foo() {
        let cond = false
        _ = cond ? Self.aaa : Self.bbb
    }
}
```

still produces ```colon``` :( in 0.34.0

**Comment by a-25** [comment]
@Lukasz2891: fix is not released yet
